bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java

```diff
@@ -342,6 +344,14 @@ public String getLanguageDirection() {
             (request != null && StringUtils.isNotBlank(request.getParameter(GuideConstants.AF_DATA_REF)))) {
             formDataEnabled = true;
         }
+
+        // set draftId in properties in case of forms portal prefill
+        if (request != null && StringUtils.isNotBlank(request.getParameter(GuideConstants.AF_DATA_REF))) {
+            final String dataRef = request.getParameter(GuideConstants.AF_DATA_REF);
+            if (dataRef.startsWith(DRAFT_PREFILL_SERVICE)) {
+                properties.put(ReservedProperties.FD_DRAFT_ID, StringUtils.substringAfter(dataRef, DRAFT_PREFILL_SERVICE));
+            }
+        }
         properties.put(FD_ROLE_ATTRIBUTE, getRoleAttribute());
         properties.put(FD_FORM_DATA_ENABLED, formDataEnabled);
         properties.put(ReservedProperties.FD_AUTO_SAVE_PROPERTY_WRAPPER, this.autoSaveConfig);
```

Comment:
Can we conditionally include the auto save configuration ? This is unnecessarily bloating the JSON unless someone explicitly add's some auto save property, only then we should add this in json